tests/core/mempool/test_mempool_manager.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+import asyncio
+from typing import Any, Iterator, List, Optional
+
+import pytest
+import pytest_asyncio
+from blspy import G2Element
+
+from chia.consensus.block_record import BlockRecord
+from chia.consensus.default_constants import DEFAULT_CONSTANTS
+from chia.full_node.mempool_manager import MempoolManager
+from chia.types.blockchain_format.classgroup import ClassgroupElement
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32, bytes100
+from chia.types.coin_record import CoinRecord
+from chia.types.coin_spend import CoinSpend
+from chia.types.condition_opcodes import ConditionOpcode
+from chia.types.spend_bundle import SpendBundle
+from chia.util.errors import ValidationError
+from chia.util.ints import uint8, uint32, uint64, uint128
+
+IDENTITY_PUZZLE = Program.to(1)
+IDENTITY_PUZZLE_HASH = IDENTITY_PUZZLE.get_tree_hash()
+
+TEST_COIN = Coin(IDENTITY_PUZZLE_HASH, IDENTITY_PUZZLE_HASH, uint64(1000000000))
+TEST_HEIGHT = uint32(1)
+TEST_TIMESTAMP = uint64(1616108400)
+TEST_BLOCK_RECORD = BlockRecord(
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    TEST_HEIGHT,
+    uint128(0),
+    uint128(0),
+    uint8(0),
+    ClassgroupElement(bytes100(b"0" * 100)),
+    None,
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    uint64(0),
+    IDENTITY_PUZZLE_HASH,
+    IDENTITY_PUZZLE_HASH,
+    uint64(0),
+    uint8(0),
+    False,
+    uint32(TEST_HEIGHT - 1),
+    TEST_TIMESTAMP,
+    None,
+    uint64(0),
+    None,
+    None,
+    None,
+    None,
+    None,
+)
+
+
+async def get_coin_record(_: bytes32) -> Optional[CoinRecord]:
```

Comment:
I think it would be great if each test ensured that this callback was called with the expected coin ID as well, and not just ignored

Reply:
I created a record for this test coin and made this callback return it from a dictionary of records.